### PR TITLE
DATA-2874 | search result item context

### PR DESCRIPTION
**Pull request by bradgarropy**
_Please merge in #9 first._

## Description
Include information about the search results in the search context.

## Related Issue
[DATA-2874](https://jira.corp.magento.com/browse/DATA-2874)

## Motivation and Context
Continuing to add contexts for Search eventing.

## How Has This Been Tested?
Modified unit tests to confirm the context can be set and retrieved successfully.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/MagentoDataLayerContextManager.ts [hunk 1]
```diff
@@ -114,6 +116,20 @@ export default class MagentoDataLayerContextManager extends MagentoDataLayerBase
     this.setContext<ReferrerUrl>(REFERRER_URL_CONTEXT, context);
   }
 
+  /**
+   * Get search results context
+   */
+  getSearch(): Search {
```
Comment (deloreyj) [inline src/MagentoDataLayerContextManager.ts:122]: We'll have a context for `searchResult` and `searchInput` yeah? Should we have two getters and setters to match like `getSearchResult` and `getSearchInput`?

File: src/index.test.ts [hunk 2]
```diff
@@ -100,6 +73,41 @@ describe("contexts", () => {
     mdl.context.setProduct(context);
     expect(mdl.context.getProduct()).toEqual(context);
   });
+
+  test("referrer url context", () => {
```
Comment (deloreyj) [inline src/index.test.ts:77]: Did you alphabetize our tests? Because if so, I love it. 😄

File: src/mocks.ts [hunk 3]
```diff
@@ -44,20 +51,29 @@ export const generateProductContext = (
...
-});
-
 export const generateReferrerUrlContext = (
   overrides?: Partial<ReferrerUrl>
 ): ReferrerUrl => ({
   referrerUrl: "https://www.test.org",
   ...overrides,
 });
 
+export const generateSearchContext = (overrides?: Partial<Search>): Search => ({
```
Comment (deloreyj) [inline src/mocks.ts:61]:
```suggestion
export const generateSearchResultContext = (overrides?: Partial<SearchResult>): SearchResult => ({
```

File: src/types/contexts.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ export const MAGENTO_EXTENSION_CONTEXT = "magento-extension-context";
 export const PAGE_OFFSET_CONTEXT = "page-offset-context";
 export const PRODUCT_CONTEXT = "product-context";
 export const REFERRER_URL_CONTEXT = "referrer-url-context";
+export const SEARCH_CONTEXT = "search-context";
```
Comment (deloreyj) [inline src/types/contexts.ts:10]:
```suggestion
export const SEARCH_RESULT_CONTEXT = "search-result-context";
```

File: src/types/contexts.ts [hunk 5]
```diff
@@ -17,6 +18,7 @@ export type ContextName =
   | typeof PAGE_OFFSET_CONTEXT
   | typeof PRODUCT_CONTEXT
   | typeof REFERRER_URL_CONTEXT
+  | typeof SEARCH_CONTEXT
```
Comment (deloreyj) [inline src/types/contexts.ts:21]:
```suggestion
  | typeof SEARCH_RESULT_CONTEXT
```

File: src/types/schemas/customUrl.ts [hunk 6]
```diff
@@ -1,3 +1,8 @@
+/*
```
Comment (deloreyj) [inline src/types/schemas/customUrl.ts:1]: Thank you!

File: src/types/schemas/search.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  title: string;
+  url: string;
+  rank: number;
+  resultType: "category" | "suggestion" | "product";
+  sku?: string;
+  imageUrl?: string;
+  price?: string;
+}
+
+export interface Search {
```
Comment (deloreyj) [inline src/types/schemas/search.ts:16]:
```suggestion
export interface SearchResult {
```

File: src/index.test.ts [hunk 8]
```diff
@@ -10,6 +10,7 @@ import {
   generatePageOffsetsContext,
   generateProductContext,
   generateReferrerUrlContext,
+  generateSearchContext,
```
Comment (deloreyj) [inline src/index.test.ts:13]:
```suggestion
  generateSearchResultContext,
```

File: src/MagentoDataLayerContextManager.ts [hunk 9]
```diff
@@ -8,6 +8,7 @@ import {
   MAGENTO_EXTENSION_CONTEXT,
   PAGE_OFFSET_CONTEXT,
   PRODUCT_CONTEXT,
+  SEARCH_CONTEXT,
```
Comment (deloreyj) [inline src/MagentoDataLayerContextManager.ts:11]:
```suggestion
  SEARCH_RESULT_CONTEXT,
```

File: src/MagentoDataLayerContextManager.ts [hunk 10]
```diff
@@ -23,6 +24,7 @@ import { StorefrontInstance } from "./types/schemas/storefrontInstance";
 import { ShoppingCart } from "./types/schemas/shoppingCart";
 import { CustomUrl } from "./types/schemas/customUrl";
 import { ReferrerUrl } from "./types/schemas/referrerUrl";
+import { Search } from "./types/schemas/search";
```
Comment (deloreyj) [inline src/MagentoDataLayerContextManager.ts:27]:
```suggestion
import { SearchResult } from "./types/schemas/search";
```

File: src/index.test.ts (hunk 2 above)
Comment (bradgarropy) [reply src/index.test.ts:77]: Hahaha yes! It helped me ensure we had a test for every context.

**Review by deloreyj — APPROVED** [review]

**Pull request merged**